src/runtime/internal/dom.ts

```diff
@@ -215,35 +215,57 @@ export function select_multiple_value(select) {
 	return [].map.call(select.querySelectorAll(':checked'), option => option.__value);
 }
 
-export function add_resize_listener(element, fn) {
-	if (getComputedStyle(element).position === 'static') {
-		element.style.position = 'relative';
+// unfortunately it can't be a constant as that wouldn't be tree-shakeable
+// so we cache the result instead
+let crossorigin: boolean;
+
+export function is_crossorigin() {
+	if (crossorigin === undefined) {
+		try {
+			if (typeof window === 'undefined' || !window.parent || window.parent.document) {
+				crossorigin = false;
```

Comment:
Is it possible that none of these conditions will be true, and yet an exception is not thrown? If yes, `crossorigin` wouldn't get set in those cases. If not, could this be made a little clearer?

Reply:
Either the last one throws, or one of them is true, but I'm not sure how to make it clearer. I suppose `crossorigin = false` could come before the tests, would that help? Perhaps adding a `void` to signify it's done on purpose instead of a forgotten statement?

```js
if (crossorigin === undefined) {
    crossorigin = false;

    try {
        if (typeof window !== 'undefined' && window.parent) {
            void window.parent.document;
        }
    } catch (error) {
        crossorigin = true;
    }
}

return crossorigin;
```